definitions/npm/react-redux_v5.x.x/flow_v0.62.0-/test_connect.js

```diff
@@ -0,0 +1,319 @@
+    render() {
+      return <div>{this.props.passtrough} {this.props.fromStateToProps}</div>;
+    }
+  }
+
+  type State = {a: number};
+  type InputProps = {
+    forMapStateToProps: string
+  };
+  const mapStateToProps = (state: State, props: InputProps) => {
+    return {
+      fromStateToProps: 'str' + state.a
+    }
+  };
+
+  const Connected = connect(mapStateToProps)(Com);
+  <Connected passtrough={123} forMapStateToProps={'data'} passtroughWithDefaultProp={123}/>;
+  // OK without passtroughWithDefaultProp
+  <Connected passtrough={123} forMapStateToProps={'data'}/>;
+  //$ExpectError wrong type for passtrough
+  <Connected passtrough={''} forMapStateToProps={'data'} passtroughWithDefaultProp={123}/>;
+  //$ExpectError wrong type for forMapStateToProps
+  <Connected passtrough={123} forMapStateToProps={321} passtroughWithDefaultProp={123}/>;
+  //$ExpectError wrong type for  passtroughWithDefaultProp
+  <Connected passtrough={123} forMapStateToProps={'data'} passtroughWithDefaultProp={''}/>;
+  //$ExpectError passtrough missing
+  <Connected forMapStateToProps={'data'} />;
+  //$ExpectError forMapStateToProps missing
+  <Connected passtrough={123}/>;
+  //$ExpectError takes in only React components
+  connect(mapStateToProps)('');
+}
+
+function testExactProps() {
+  type Props = {|
+    forMapStateToProps: string,
+    passtrough: number,
+    fromStateToProps: string
+  |};
+
+  class Com extends React.Component<Props> {
+    render() {
+      return <div>{this.props.passtrough} {this.props.fromStateToProps}</div>;
+    }
+  }
+
+  type State = {a: number};
+  type InputProps = {|
+    forMapStateToProps: string,
+    passtrough: number,
+  |};
+
+  const mapStateToProps = (state: State, props: InputProps) => {
+    return {
+      fromStateToProps: 'str' + state.a
+    }
+  };
+
+  const Connected = connect(mapStateToProps)(Com);
+  <Connected passtrough={123} forMapStateToProps={'data'} />;
+  //$ExpectError extra prop what exact props does not allow
+  <Connected passtrough={123} forMapStateToProps={321} extraProp={123}/>;
+  //$ExpectError wrong type for forMapStateToProps
+  <Connected passtrough={123} forMapStateToProps={321}/>;
+  //$ExpectError passtrough missing
+  <Connected forMapStateToProps={'data'} />;
+  //$ExpectError forMapStateToProps missing
+  <Connected passtrough={123}/>;
+  //$ExpectError takes in only React components
+  connect(mapStateToProps)('');
+}
+
+function testWithStatelessFunctionalComponent() {
+  type Props = {passtrough: number, fromStateToProps: string};
+  const Com = (props: Props) => <div>{props.passtrough} {props.fromStateToProps}</div>
+
+  type State = {a: number};
+  type InputProps = {
+    forMapStateToProps: string
+  };
+  const mapStateToProps = (state: State, props: InputProps) => {
+    return {
+      fromStateToProps: 'str' + state.a
+    }
+  };
+
+  const Connected = connect(mapStateToProps)(Com);
+  <Connected passtrough={123} forMapStateToProps={'data'}/>;
```

Comment:
Just a minor thing, but I suspect that `passtrough` is a typo for `passthrough`.

Reply:
![giphy](https://user-images.githubusercontent.com/12481532/36702153-c9541ab2-1bba-11e8-94c7-da7e8a260c85.gif)